server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/akka/actors/io/EncDecActorMessageProcessor.java

```diff
@@ -113,13 +114,18 @@ void encodeAndReply(SessionResponse message) {
             sessionResponseMeter.mark();
             processSessionResponse(message);
         } catch (Exception e) {
-            processErrors(message.getChannelContext(), message.getErrorConverter(), e);
+            processErrors(message.getChannelContext(), message.getErrorBuilder(), e);
         }
     }
 
     public void forward(ActorContext context, SessionAware message) {
-        LOG.debug("Forwarding session aware message: {}", message);
-        this.opsActor.tell(message, context.self());
+        if (isApplicationTokenValid(message.getSessionInfo().getApplicationToken())) {
```

Comment:
Let's completely remove this validation for SessionAware messages and keep it only for SessionInit messages for performance reasons